zebra-state/src/error.rs

```diff
@@ -106,30 +106,58 @@ impl<E: std::error::Error + 'static> From<BoxError> for LayeredStateError<E> {
     }
 }
 
-/// An error describing the reason a block or its descendants could not be reconsidered after
-/// potentially being invalidated from the chain_set.
+/// An error describing why a `InvalidateBlock` request failed.
 #[derive(Debug, Error)]
+#[non_exhaustive]
+pub enum InvalidateError {
+    #[error("cannot invalidate blocks while still committing checkpointed blocks")]
+    /// The state is currently checkpointing blocks and cannot accept invalidation requests.
+    CannotInvalidateWhileCheckpointing,
+
+    #[error("failed to send invalidate block request to block write task")]
+    /// Sending the invalidate request to the block write task failed.
+    SendInvalidateRequestFailed,
+
+    #[error("invalidate block request was unexpectedly dropped")]
+    /// The invalidate request was dropped before processing.
+    InvalidateRequestDropped,
+
+    #[error("contextual validation of the block failed")]
+    /// The block failed contextual validation during invalidation.
+    ValidationError(#[from] ValidateContextError),
```

Comment:
We always want to put attributes immediately above the item to which they're applied.

This applies to the updates to `ReconsiderError` below as well.

We also don't need the `ValidationError` variant here wrapping a `ValidateContextError` as there's only one possible `ValidateContextError` that it could be, and it's not really a validation error, the block is just missing which is only relevant to the invalidate block request.

```suggestion
    /// The state is currently checkpointing blocks and cannot accept invalidation requests.
    #[error("cannot invalidate blocks while still committing checkpointed blocks")]
    ProcessingCheckpointedBlocks,

    /// Sending the invalidate request to the block write task failed.
    #[error("failed to send invalidate block request to block write task")]
    SendInvalidateRequestFailed,

    /// The invalidate request was dropped before processing.
    #[error("invalidate block request was unexpectedly dropped")]
    InvalidateRequestDropped,

    #[error("block hash {0} not found in any non-finalized chain")]
    BlockNotFound(block::Hash),
```